src/ORM/Table.php

```diff
@@ -530,10 +531,40 @@ public function setSchema($schema)
         }
 
         $this->_schema = $schema;
+        $this->validateSchema();
 
         return $this;
     }
 
+    /**
+     * Validates the identifiers are valid for generating queries.
+     *
+     * @return void
+     * @throws \RuntimeException When identifiers are invalid
+     */
+    protected function validateSchema(): void
+    {
+        if ($this->_schema === null) {
+            throw new RuntimeException("Unable to validate `{$this->getAlias()}` without schema.");
+        }
+
+        $maxLength = null;
+        if (method_exists($this->getConnection()->getDriver(), "getMaxAliasLength")) {
+            $maxLength = $this->getConnection()->getDriver()->getMaxAliasLength();
+        }
+
+        if ($maxLength) {
+            $table = $this->getAlias();
+            foreach ($this->_schema->columns() as $name) {
+                if (strlen($table . '__' . $name) > $maxLength) {
```

Comment:
This feels like a lot of unnecessary overheard. I would suggest simply skipping calling this method when debug is turned off.

Reply:
Agreed. This is the minimal place I found to check, but I didn't want to try caching the result in the schema.